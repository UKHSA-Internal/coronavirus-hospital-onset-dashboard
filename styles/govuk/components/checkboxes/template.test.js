/**
 * @jest-environment jsdom
 */
/* eslint-env jest */

const axe = require('../../../../lib/axe-helper')

const { render, getExamples, htmlWithClassName } = require('../../../../lib/jest-helpers')

const examples = getExamples('checkboxes')

const WORD_BOUNDARY = '\\b'
const WHITESPACE = '\\s'

describe('Checkboxes', () => {
  it('default example passes accessibility tests', async () => {
    const $ = render('checkboxes', examples.default)

    const results = await axe($.html())
    expect(results).toHaveNoViolations()
  })

  it('render example with minimum required name and items', () => {
    const $ = render('checkboxes', {
      name: 'example-name',
      items: [
        {
          value: '1',
          text: 'Option 1'
        },
        {
          value: '2',
          text: 'Option 2'
        }
      ]
    })

    const $component = $('.govuk-checkboxes')

    const $firstInput = $component.find('.govuk-checkboxes__item:first-child input')
    const $firstLabel = $component.find('.govuk-checkboxes__item:first-child label')
    expect($firstInput.attr('name')).toEqual('example-name')
    expect($firstInput.val()).toEqual('1')
    expect($firstLabel.text()).toContain('Option 1')

    const $lastInput = $component.find('.govuk-checkboxes__item:last-child input')
    const $lastLabel = $component.find('.govuk-checkboxes__item:last-child label')
    expect($lastInput.attr('name')).toEqual('example-name')
    expect($lastInput.val()).toEqual('2')
    expect($lastLabel.text()).toContain('Option 2')
  })

  it('render example without falsely values', () => {
    const $ = render('checkboxes', {
      name: 'example-name',
      items: [
        {
          value: '1',
          text: 'Option 1'
        },
        undefined,
        null,
        false,
        {
          value: '2',
          text: 'Option 2'
        }
      ]
    })

    const $component = $('.govuk-checkboxes')
    const $items = $component.find('.govuk-checkboxes__item')

    expect($items.length).toEqual(2)
  })

  it('render classes', () => {
    const $ = render('checkboxes', {
      name: 'example-name',
      items: [
        {
          value: '1',
          text: 'Option 1'
        },
        {
          value: '2',
          text: 'Option 2'
        }
      ],
      classes: 'app-checkboxes--custom-modifier'
    })

    const $component = $('.govuk-checkboxes')

    expect($component.hasClass('app-checkboxes--custom-modifier')).toBeTruthy()
  })

  it('renders initial aria-describedby on fieldset', () => {
    const describedById = 'some-id'

    const $ = render('checkboxes', {
      name: 'example-name',
      fieldset: {
        describedBy: describedById
      },
      items: [
        {
          value: '1',
          text: 'Option 1'
        },
        {
          value: '2',
          text: 'Option 2'
        }
      ]
    })

    const $fieldset = $('.govuk-fieldset')
    expect($fieldset.attr('aria-describedby')).toMatch(describedById)
  })

  it('render attributes', () => {
    const $ = render('checkboxes', {
      name: 'example-name',
      items: [
        {
          value: '1',
          text: 'Option 1'
        },
        {
          value: '2',
          text: 'Option 2'
        }
      ],
      attributes: {
        'data-attribute': 'value',
        'data-second-attribute': 'second-value'
      }
    })

    const $component = $('.govuk-checkboxes')

    expect($component.attr('data-attribute')).toEqual('value')
    expect($component.attr('data-second-attribute')).toEqual('second-value')
  })

  it('renders with a form group wrapper', () => {
    const $ = render('checkboxes', {})

    const $formGroup = $('.govuk-form-group')
    expect($formGroup.length).toBeTruthy()
  })

  it('render a custom class on the form group', () => {
    const $ = render('checkboxes', {
      name: 'example-name',
      items: [
        {
          value: '1',
          text: 'Option 1'
        },
        {
          value: '2',
          text: 'Option 2'
        }
      ],
      formGroup: {
        classes: 'custom-group-class'
      }
    })

    const $formGroup = $('.govuk-form-group')
    expect($formGroup.hasClass('custom-group-class')).toBeTruthy()
  })

  describe('items', () => {
    it('render a matching label and input using name by default', () => {
      const $ = render('checkboxes', {
        name: 'example-name',
        items: [
          {
            value: '1',
            text: 'Option 1'
          },
          {
            value: '2',
            text: 'Option 2'
          }
        ]
      })

      const $component = $('.govuk-checkboxes')

      const $firstInput = $component.find('.govuk-checkboxes__item:first-child input')
      const $firstLabel = $component.find('.govuk-checkboxes__item:first-child label')
      expect($firstInput.attr('id')).toEqual('example-name')
      expect($firstLabel.attr('for')).toEqual('example-name')

      const $lastInput = $component.find('.govuk-checkboxes__item:last-child input')
      const $lastLabel = $component.find('.govuk-checkboxes__item:last-child label')
      expect($lastInput.attr('id')).toEqual('example-name-2')
      expect($lastLabel.attr('for')).toEqual('example-name-2')
    })

    it('render a matching label and input using custom idPrefix', () => {
      const $ = render('checkboxes', {
        idPrefix: 'custom',
        name: 'example-name',
        items: [
          {
            value: '1',
            text: 'Option 1'
          },
          {
            value: '2',
            text: 'Option 2'
          }
        ]
      })

      const $component = $('.govuk-checkboxes')

      const $firstInput = $component.find('.govuk-checkboxes__item:first-child input')
      const $firstLabel = $component.find('.govuk-checkboxes__item:first-child label')
      expect($firstInput.attr('id')).toEqual('custom')
      expect($firstLabel.attr('for')).toEqual('custom')

      const $lastInput = $component.find('.govuk-checkboxes__item:last-child input')
      const $lastLabel = $component.find('.govuk-checkboxes__item:last-child label')
      expect($lastInput.attr('id')).toEqual('custom-2')
      expect($lastLabel.attr('for')).toEqual('custom-2')
    })

    it('render explicitly passed item ids', () => {
      const $ = render('checkboxes', {
        name: 'example-name',
        items: [
          {
            value: '1',
            text: 'Option 1'
          },
          {
            id: 'custom_item_id',
            value: '2',
            text: 'Option 2'
          }
        ]
      })

      const $component = $('.govuk-checkboxes')

      const $firstInput = $component.find('.govuk-checkboxes__item:first-child input')
      expect($firstInput.attr('id')).toBe('example-name')

      const $lastInput = $component.find('.govuk-checkboxes__item:last-child input')
      const $lastLabel = $component.find('.govuk-checkboxes__item:last-child label')
      expect($lastInput.attr('id')).toBe('custom_item_id')
      expect($lastLabel.attr('for')).toEqual('custom_item_id')
    })

    it('render explicitly passed item names', () => {
      const $ = render('checkboxes', {
        name: 'example-name',
        items: [
          {
            value: '1',
            text: 'Option 1',
            name: 'custom-item-name-1'
          },
          {
            value: '2',
            text: 'Option 2',
            name: 'custom-item-name-2'
          }
        ]
      })

      const $component = $('.govuk-checkboxes')

      const $firstInput = $component.find('.govuk-checkboxes__item:first-child input')
      expect($firstInput.attr('name')).toBe('custom-item-name-1')

      const $lastInput = $component.find('.govuk-checkboxes__item:last-child input')
      expect($lastInput.attr('name')).toBe('custom-item-name-2')
    })

    it('render disabled', () => {
      const $ = render('checkboxes', {
        name: 'example-name',
        items: [
          {
            value: '1',
            text: 'Option 1',
            disabled: true
          },
          {
            value: '2',
            text: 'Option 2'
          }
        ]
      })

      const $component = $('.govuk-checkboxes')

      const $firstInput = $component.find('.govuk-checkboxes__item:first-child input')
      expect($firstInput.attr('disabled')).toEqual('disabled')
    })

    it('render checked', () => {
      const $ = render('checkboxes', {
        name: 'example-name',
        items: [
          {
            value: '1',
            text: 'Option 1'
          },
          {
            value: '2',
            text: 'Option 2',
            checked: true
          },
          {
            value: '3',
            text: 'Option 3',
            checked: true
          }
        ]
      })

      const $component = $('.govuk-checkboxes')
      const $secondInput = $component.find('.govuk-checkboxes__item:nth-child(2) input')
      const $lastInput = $component.find('.govuk-checkboxes__item:last-child input')
      expect($secondInput.attr('checked')).toEqual('checked')
      expect($lastInput.attr('checked')).toEqual('checked')
    })

    describe('when they include attributes', () => {
      it('it renders the attributes', () => {
        const $ = render('checkboxes', {
          name: 'example-name',
          items: [
            {
              value: '1',
              text: 'Option 1',
              attributes: {
                'data-attribute': 'ABC',
                'data-second-attribute': 'DEF'
              }
            },
            {
              value: '2',
              text: 'Option 2',
              attributes: {
                'data-attribute': 'GHI',
                'data-second-attribute': 'JKL'
              }
            }
          ]
        })

        const $component = $('.govuk-checkboxes')

        const $firstInput = $component.find('.govuk-checkboxes__item:first-child input')
        expect($firstInput.attr('data-attribute')).toEqual('ABC')
        expect($firstInput.attr('data-second-attribute')).toEqual('DEF')

        const $lastInput = $component.find('.govuk-checkboxes__item:last-child input')
        expect($lastInput.attr('data-attribute')).toEqual('GHI')
        expect($lastInput.attr('data-second-attribute')).toEqual('JKL')
      })
    })
  })

  describe('when they include a hint', () => {
    it('it renders the hint text', () => {
      const $ = render('checkboxes', {
        name: 'gov',
        items: [
          {
            value: 'value',
            text: 'This is text',
            hint: {
              text: 'This is a hint'
            }
          }
        ]
      })
      expect($('.govuk-checkboxes__hint').text()).toContain('This is a hint')
    })

    it('it renders the correct id attribute for the hint', () => {
      const $ = render('checkboxes', {
        name: 'gov',
        items: [
          {
            value: 'value',
            text: 'This is text',
            id: 'item-id',
            hint: {
              text: 'This is a hint'
            }
          }
        ]
      })

      expect($('.govuk-checkboxes__hint').attr('id')).toBe('item-id-item-hint')
    })

    it('the input describedBy attribute matches the item hint id', () => {
      const $ = render('checkboxes', {
        name: 'gov',
        items: [
          {
            value: 'value',
            text: 'This is text',
            id: 'item-id',
            hint: {
              text: 'This is a hint'
            }
          }
        ]
      })

      expect($('.govuk-checkboxes__input').attr('aria-describedby')).toBe('item-id-item-hint')
    })
  })

  describe('render conditionals', () => {
    it('hidden by default when not checked', () => {
      const $ = render('checkboxes', {
        name: 'example-conditional',
        items: [
          {
            value: 'yes',
            text: 'Yes',
            conditional: {
              html: 'Conditional content that should be hidden'
            }
          },
          {
            value: 'no',
            text: 'No'
          }
        ]
      })

      const $component = $('.govuk-checkboxes')

      const $firstConditional = $component.find('.govuk-checkboxes__conditional').first()
      expect($firstConditional.html()).toContain('Conditional content that should be hidden')
      expect($firstConditional.hasClass('govuk-checkboxes__conditional--hidden')).toBeTruthy()
    })
    it('visible by default when checked', () => {
      const $ = render('checkboxes', {
        name: 'example-conditional',
        items: [
          {
            value: 'yes',
            text: 'Yes',
            checked: true,
            conditional: {
              html: 'Conditional content that should be visible'
            }
          },
          {
            value: 'no',
            text: 'No'
          }
        ]
      })

      const $component = $('.govuk-checkboxes')

      const $firstConditional = $component.find('.govuk-checkboxes__conditional').first()
      expect($firstConditional.html()).toContain('Conditional content that should be visible')
      expect($firstConditional.hasClass('govuk-checkboxes__conditional--hidden')).toBeFalsy()
    })
    it('with association to the input they are controlled by', () => {
      const $ = render('checkboxes', {
        name: 'example-conditional',
        items: [
          {
            value: 'yes',
            text: 'Yes',
            conditional: {
              html: 'Conditional content that should be hidden'
            }
          },
          {
            value: 'no',
            text: 'No'
          }
        ]
      })

      const $component = $('.govuk-checkboxes')

      const $firstInput = $component.find('.govuk-checkboxes__input').first()
      const $firstConditional = $component.find('.govuk-checkboxes__conditional').first()

      expect($firstInput.attr('data-aria-controls')).toBe('conditional-example-conditional')
      expect($firstConditional.attr('id')).toBe('conditional-example-conditional')
    })

    it('omits empty conditionals', () => {
      const $ = render('checkboxes', {
        name: 'example-conditional',
        items: [
          {
            value: 'foo',
            text: 'Foo',
            conditional: {
              html: false
            }
          }
        ]
      })

      const $component = $('.govuk-checkboxes')
      expect($component.find('.govuk-checkboxes__conditional').length).toEqual(0)
    })

    it('does not associate checkboxes with empty conditionals', () => {
      const $ = render('checkboxes', {
        name: 'example-conditional',
        items: [
          {
            value: 'foo',
            text: 'Foo',
            conditional: {
              html: false
            }
          }
        ]
      })

      const $input = $('.govuk-checkboxes__input').first()
      expect($input.attr('data-aria-controls')).toBeFalsy()
    })
  })

  it('render additional label classes', () => {
    const $ = render('checkboxes', {
      name: 'example-label-classes',
      items: [
        {
          value: 'yes',
          text: 'Yes',
          label: {
            classes: 'bold'
          }
        }
      ]
    })

    const $component = $('.govuk-checkboxes')
    const $label = $component.find('.govuk-checkboxes__item label')
    expect($label.hasClass('bold')).toBeTruthy()
  })

  describe('when they include an error message', () => {
    it('renders the error message', () => {
      const $ = render('checkboxes', {
        name: 'example',
        errorMessage: {
          text: 'Please select an option'
        }
      })

      expect(htmlWithClassName($, '.govuk-error-message')).toMatchSnapshot()
    })

    it('uses the idPrefix for the error message id if provided', () => {
      const $ = render('checkboxes', {
        name: 'name-of-checkboxes',
        errorMessage: {
          text: 'Please select an option'
        },
        idPrefix: 'id-prefix',
        items: [
          {
            value: 'animal',
            text: 'Waste from animal carcasses'
          }
        ]
      })

      const $errorMessage = $('.govuk-error-message')

      expect($errorMessage.attr('id')).toEqual('id-prefix-error')
    })

    it('falls back to using the name for the error message id', () => {
      const $ = render('checkboxes', {
        name: 'name-of-checkboxes',
        errorMessage: {
          text: 'Please select an option'
        },
        items: [
          {
            value: 'animal',
            text: 'Waste from animal carcasses'
          }
        ]
      })

      const $errorMessage = $('.govuk-error-message')

      expect($errorMessage.attr('id')).toEqual('name-of-checkboxes-error')
    })

    it('associates the fieldset as "described by" the error message', () => {
      const $ = render('checkboxes', examples['with fieldset and error message'])

      const $fieldset = $('.govuk-fieldset')
      const $errorMessage = $('.govuk-error-message')

      const errorMessageId = new RegExp(
        WORD_BOUNDARY + $errorMessage.attr('id') + WORD_BOUNDARY
      )

      expect($fieldset.attr('aria-describedby'))
        .toMatch(errorMessageId)
    })

    it('associates the fieldset as "described by" the error message and parent fieldset', () => {
      const describedById = 'some-id'
      const params = examples['with fieldset and error message']

      params.fieldset.describedBy = describedById

      const $ = render('checkboxes', params)

      const $fieldset = $('.govuk-fieldset')
      const $errorMessage = $('.govuk-error-message')

      const errorMessageId = new RegExp(
        WORD_BOUNDARY + describedById + WHITESPACE + $errorMessage.attr('id') + WORD_BOUNDARY
      )

      expect($fieldset.attr('aria-describedby'))
        .toMatch(errorMessageId)

      delete params.fieldset.describedBy
    })

    it('does not associate each input as "described by" the error message', () => {
      const $ = render('checkboxes', examples['with error message and hints on items'])

      const $inputs = $('input')

      $inputs.each((index, input) => {
        let expectedDescribedById = `waste-${(index + 1)}-item-hint`
        if (index === 0) {
          expectedDescribedById = `waste-item-hint`
        }
        expect($(input).attr('aria-describedby')).toEqual(expectedDescribedById)
      })
    })

    it('renders with a form group wrapper that has an error state', () => {
      const $ = render('checkboxes', {
        errorMessage: {
          text: 'Error message'
        }
      })

      const $formGroup = $('.govuk-form-group')
      expect($formGroup.hasClass('govuk-form-group--error')).toBeTruthy()
    })
  })

  describe('when they include a hint', () => {
    it('renders the hint', () => {
      const $ = render('checkboxes', {
        name: 'example',
        hint: {
          text: 'If you have dual nationality, select all options that are relevant to you.'
        },
        items: [
          {
            value: 'british',
            text: 'British',
            hint: {
              text: 'Hint for british option here'
            }
          },
          {
            value: 'irish',
            text: 'Irish'
          },
          {
            hint: {
              text: 'Hint for other option here',
              classes: 'app-checkboxes__hint-other',
              attributes: {
                'data-test-attribute': true
              }
            }
          }
        ]
      })

      expect(htmlWithClassName($, '.govuk-hint')).toMatchSnapshot()
    })

    it('associates the fieldset as "described by" the hint', () => {
      const $ = render('checkboxes', {
        name: 'example',
        fieldset: {
          legend: {
            text: 'What is your nationality?'
          }
        },
        hint: {
          text: 'If you have dual nationality, select all options that are relevant to you.'
        }
      })

      const $fieldset = $('.govuk-fieldset')
      const $hint = $('.govuk-hint')

      const hintId = new RegExp(
        WORD_BOUNDARY + $hint.attr('id') + WORD_BOUNDARY
      )

      expect($fieldset.attr('aria-describedby')).toMatch(hintId)
    })

    it('associates the fieldset as "described by" the hint and parent fieldset', () => {
      const describedById = 'some-id'

      const $ = render('checkboxes', {
        name: 'example',
        fieldset: {
          describedBy: describedById,
          legend: {
            text: 'What is your nationality?'
          }
        },
        hint: {
          text: 'If you have dual nationality, select all options that are relevant to you.'
        }
      })
      const $fieldset = $('.govuk-fieldset')
      const $hint = $('.govuk-hint')

      const hintId = new RegExp(
        WORD_BOUNDARY + describedById + WHITESPACE + $hint.attr('id') + WORD_BOUNDARY
      )

      expect($fieldset.attr('aria-describedby')).toMatch(hintId)
    })
  })

  describe('when they include both a hint and an error message', () => {
    it('associates the fieldset as described by both the hint and the error message', () => {
      const $ = render('checkboxes', {
        name: 'example',
        errorMessage: {
          text: 'Please select an option'
        },
        fieldset: {
          legend: {
            text: 'What is your nationality?'
          }
        },
        hint: {
          text: 'If you have dual nationality, select all options that are relevant to you.'
        }
      })

      const $fieldset = $('.govuk-fieldset')
      const errorMessageId = $('.govuk-error-message').attr('id')
      const hintId = $('.govuk-hint').attr('id')

      const combinedIds = new RegExp(
        WORD_BOUNDARY + hintId + WHITESPACE + errorMessageId + WORD_BOUNDARY
      )

      expect($fieldset.attr('aria-describedby'))
        .toMatch(combinedIds)
    })

    it('associates the fieldset as described by the hint, error message and parent fieldset', () => {
      const describedById = 'some-id'

      const $ = render('checkboxes', {
        name: 'example',
        errorMessage: {
          text: 'Please select an option'
        },
        fieldset: {
          describedBy: describedById,
          legend: {
            text: 'What is your nationality?'
          }
        },
        hint: {
          text: 'If you have dual nationality, select all options that are relevant to you.'
        }
      })

      const $fieldset = $('.govuk-fieldset')
      const errorMessageId = $('.govuk-error-message').attr('id')
      const hintId = $('.govuk-hint').attr('id')

      const combinedIds = new RegExp(
        WORD_BOUNDARY + describedById + WHITESPACE + hintId + WHITESPACE + errorMessageId + WORD_BOUNDARY
      )

      expect($fieldset.attr('aria-describedby'))
        .toMatch(combinedIds)
    })
  })

  describe('nested dependant components', () => {
    it('have correct nesting order', () => {
      const $ = render('checkboxes', {
        fieldset: {
          legend: {
            text: 'What is your nationality?'
          }
        }
      })

      const $component = $('.govuk-form-group > .govuk-fieldset > .govuk-checkboxes')
      expect($component.length).toBeTruthy()
    })

    it('passes through label params without breaking', () => {
      const $ = render('checkboxes', {
        name: 'example-name',
        items: [
          {
            value: '1',
            html: '<b>Option 1</b>',
            label: {
              attributes: {
                'data-attribute': 'value',
                'data-second-attribute': 'second-value'
              }
            }
          },
          {
            value: '2',
            text: '<b>Option 2</b>',
            checked: true
          }
        ]
      })

      expect(htmlWithClassName($, '.govuk-checkboxes__label')).toMatchSnapshot()
    })

    it('passes through fieldset params without breaking', () => {
      const $ = render('checkboxes', {
        name: 'example',
        errorMessage: {
          text: 'Please select an option'
        },
        fieldset: {
          classes: 'app-fieldset--custom-modifier',
          attributes: {
            'data-attribute': 'value',
            'data-second-attribute': 'second-value'
          },
          legend: {
            text: 'What is your nationality?'
          }
        },
        hint: {
          text: 'If you have dual nationality, select all options that are relevant to you.'
        }
      })

      expect(htmlWithClassName($, '.govuk-fieldset')).toMatchSnapshot()
    })

    it('passes through html fieldset params without breaking', () => {
      const $ = render('checkboxes', {
        name: 'example-name',
        items: [
          {
            value: 'british',
            text: 'British'
          },
          {
            value: 'irish',
            text: 'Irish'
          }
        ],
        fieldset: {
          legend: {
            html: 'What is your <b>nationality</b>?'
          }
        }
      })

      expect(htmlWithClassName($, '.govuk-fieldset')).toMatchSnapshot()
    })
  })

  describe('single checkbox without a fieldset', () => {
    it('adds aria-describedby to input if there is an error', () => {
      const $ = render('checkboxes', examples["with single option set 'aria-describedby' on input"])
      const $input = $('input')
      expect($input.attr('aria-describedby')).toMatch('t-and-c-error')
    })

    it('adds aria-describedby to input if there is an error and parent fieldset', () => {
      const describedById = 'some-id'
      const params = examples["with single option set 'aria-describedby' on input"]

      params.describedBy = describedById

      const $ = render('checkboxes', params)
      const $input = $('input')

      expect($input.attr('aria-describedby'))
        .toMatch(`${describedById} t-and-c-error`)

      delete params.describedBy
    })
  })

  describe('single checkbox (with hint) without a fieldset', () => {
    it('adds aria-describedby to input if there is an error and a hint', () => {
      const $ = render('checkboxes', examples["with single option (and hint) set 'aria-describedby' on input"])
      const $input = $('input')
      expect($input.attr('aria-describedby')).toMatch('t-and-c-with-hint-error t-and-c-with-hint-item-hint')
    })

    it('adds aria-describedby to input if there is an error, hint and parent fieldset', () => {
      const describedById = 'some-id'
      const params = examples["with single option (and hint) set 'aria-describedby' on input"]

      params.describedBy = describedById

      const $ = render('checkboxes', params)
      const $input = $('input')

      expect($input.attr('aria-describedby'))
        .toMatch(`${describedById} t-and-c-with-hint-error t-and-c-with-hint-item-hint`)
    })
  })
})
