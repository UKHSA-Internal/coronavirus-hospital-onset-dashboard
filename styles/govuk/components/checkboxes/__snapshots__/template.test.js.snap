// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkboxes nested dependant components passes through fieldset params without breaking 1`] = `
<fieldset class="govuk-fieldset app-fieldset--custom-modifier"
          aria-describedby="example-hint example-error"
          data-attribute="value"
          data-second-attribute="second-value"
>
  <legend class="govuk-fieldset__legend">
    What is your nationality?
  </legend>
</fieldset>
`;

exports[`Checkboxes nested dependant components passes through html fieldset params without breaking 1`] = `
<fieldset class="govuk-fieldset">
  <legend class="govuk-fieldset__legend">
    What is your
    <b>
      nationality
    </b>
    ?
  </legend>
</fieldset>
`;

exports[`Checkboxes nested dependant components passes through label params without breaking 1`] = `
<label class="govuk-label govuk-checkboxes__label"
       data-attribute="value"
       data-second-attribute="second-value"
       for="example-name"
>
  <b>
    Option 1
  </b>
</label>
<label class="govuk-label govuk-checkboxes__label"
       for="example-name-2"
>
  &lt;b&gt;Option 2&lt;/b&gt;
</label>
`;

exports[`Checkboxes when they include a hint renders the hint 1`] = `
<span id="example-hint"
      class="govuk-hint"
>
  If you have dual nationality, select all options that are relevant to you.
</span>
<span id="example-item-hint"
      class="govuk-hint govuk-checkboxes__hint"
>
  Hint for british option here
</span>
<span id="example-3-item-hint"
      class="govuk-hint govuk-checkboxes__hint app-checkboxes__hint-other"
      data-test-attribute="true"
>
  Hint for other option here
</span>
`;

exports[`Checkboxes when they include an error message renders the error message 1`] = `
<span id="example-error"
      class="govuk-error-message"
>
  Please select an option
</span>
`;
